File: samples/Sample.swift
import Foundation

let maxRetries = 3
let version = "1.0.0"
let pi = 3.14159

struct Color: CustomStringConvertible {
    let r: Int, g: Int, b: Int

    var description: String {
        String(format: "#%02X%02X%02X", r, g, b)
    }

    static func fromHex(_ hex: String) -> Color {
        let hex = hex.trimmingCharacters(in: CharacterSet(charactersIn: "#"))
        let r = Int(hex.prefix(2), radix: 16) ?? 0
        let g = Int(hex.dropFirst(2).prefix(2), radix: 16) ?? 0
        let b = Int(hex.dropFirst(4).prefix(2), radix: 16) ?? 0
        return Color(r: r, g: g, b: b)
    }
}

class Theme {
    let name: String
    var colors: [String: Color] = [:]

    init(name: String) {
        self.name = name
    }

    var isDark: Bool { true }

    func addColor(_ key: String, color: Color) {
        colors[key] = color
    }

    func getColor(_ key: String) -> Color? {
        colors[key]
    }
}

@available(macOS 12.0, *)
actor ThemeManager {
    private var themes: [String: Theme] = [:]

    func register(_ theme: Theme) {
        themes[theme.name] = theme
    }

    func theme(named name: String) -> Theme? {
        themes[name]
    }
}

func processItems(_ items: [String], prefix: String = "item") -> [String] {
    items.enumerated().map { index, item in
        "\(prefix)_\(index): \(item)"
    }
}

// Enum with associated values
enum ThemeError: Error {
    case notFound(String)
    case invalidFormat(String)
}

// Generic function
func applyTheme<T: ThemeProtocol>(_ theme: T) throws where T: CustomStringConvertible {
    print("Applying theme: \(theme.description)")
}

let theme = Theme(name: "Islands Dark")
theme.addColor("keyword", color: Color.fromHex("#CF8E6D"))
theme.addColor("background", color: Color(r: 25, g: 26, b: 28))

let items = processItems(["alpha", "beta", "gamma"])
print("Theme: \(theme.name)")

File: samples/Sample.cpp
#include <iostream>
#include <vector>
#include <string>
#include <memory>
#include <algorithm>

#define MAX_SIZE 1024
#define VERSION "2.0"

constexpr double PI = 3.14159265;

namespace islands {
namespace dark {

class Color {
public:
    Color(int r, int g, int b) : r_(r), g_(g), b_(b) {}

    virtual ~Color() = default;

    std::string toHex() const {
        char buf[8];
        snprintf(buf, sizeof(buf), "#%02X%02X%02X", r_, g_, b_);
        return std::string(buf);
    }

    static Color fromHex(const std::string& hex) {
        unsigned int r, g, b;
        sscanf(hex.c_str(), "#%02x%02x%02x", &r, &g, &b);
        return Color(r, g, b);
    }

protected:
    int r_, g_, b_;
};

class Theme {
public:
    Theme(std::string name) : name_(std::move(name)) {}

    void addColor(const std::string& key, const Color& color) {
        colors_[key] = color;
    }

    Color getColor(const std::string& key) const {
        auto it = colors_.find(key);
        if (it != colors_.end()) {
            return it->second;
        }
        return Color(0, 0, 0);
    }

    void printInfo() const {
        std::cout << "Theme: " << name_ << std::endl;
        std::cout << "Colors: " << colors_.size() << std::endl;
    }

private:
    std::string name_;
    std::map<std::string, Color> colors_;
};

}  // namespace dark
}  // namespace islands

int main() {
    islands::dark::Theme theme("Islands Dark");
    theme.addColor("background", islands::dark::Color(25, 26, 28));
    theme.addColor("keyword", islands::dark::Color::fromHex("#CF8E6D"));

    theme.printInfo();
    return 0;
}